File: largest-unique-number/largest-unique-number.js
/**
 * @param {number[]} A
 * @return {number}
 */
var largestUniqueNumber = function(A) {
    map = {}
    for(let i of A){
        if (!map.hasOwnProperty(i)){
            map[i] = 1
        }else{
            map[i]++
        }
    }
    x = Object.keys(map).map(x=>parseInt(x)).filter(x => map[x] === 1)
    return x.length ? Math.max(...x) : -1
};

File: largest-unique-number/largest-unique-number.java
class Solution {
    public int largestUniqueNumber(int[] A) {
        
        Map<Integer, Integer> count = new HashMap<Integer, Integer>();
        int max = Integer.MIN_VALUE;
        
        for (int i : A){
            int x = (count.containsKey(i)) ? -1 : i;
            count.put(i, x);
        }
        
        for (int i : count.values()){
            max = Math.max(i, max);
        }
        return max;
        
        
    }
}